geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/AvroIngestProcessor.scala

```diff
@@ -47,13 +47,13 @@ object AvroIngestProcessor {
         // Schemas match, so use a regular writer.
         // TODO:  Discuss if the useProvidedFid should be true!
         (fw: FeatureWriter[SimpleFeatureType, SimpleFeature], sf: SimpleFeature) =>
-          FeatureUtils.write(fw, sf, true)
+          FeatureUtils.write(fw, sf)
       case Failure(error) =>
         if (matchMode == LenientMatch) {
           val sfConverter = convert(sft1, sft2)
```

Comment:
It's possible you could switch the order somewhere in here instead...? Didn't do the most thorough check (logically I'm not convinced it should work) but `testAvroIngestByName` ran fine with changing this to `convert(sft2, sft1)`

Also, I forgot to run **all** tests with the changes I suggested in the other thread, and turns out the changes to the sft attribute order messes up another test. Seems like I might have to let you tackle the fixing part of this :smile: